client/keystore/src/lib.rs

```diff
@@ -276,8 +298,65 @@ impl Store {
 }
 
 impl BareCryptoStore for Store {
+	fn supported_keys(
+		&self,
+		id: KeyTypeId,
+		keys: Vec<CryptoTypePublicPair>
+	) -> std::result::Result<HashSet<CryptoTypePublicPair>, TraitError> {
+		let keys = keys.into_iter().collect::<HashSet<_>>();
+		let all_keys = self.keys(id)?;
+
+		Ok(keys.intersection(&all_keys).cloned().collect())
+	}
+
+	fn keys(&self, id: KeyTypeId) -> std::result::Result<HashSet<CryptoTypePublicPair>, TraitError> {
+		let ed25519_existing_keys = self
+			.public_keys_by_type::<ed25519::Public>(id)
+			.map_err(|e| TraitError::from(e))?
+			.into_iter()
+			.map(|k| (ed25519::ED25519_CRYPTO_ID, k.to_raw_vec()));
+
+		let sr25519_existing_keys = self
+			.public_keys_by_type::<sr25519::Public>(id)
+			.map_err(|e| TraitError::from(e))?
+			.into_iter()
+			.map(|k| (sr25519::SR25519_CRYPTO_ID, k.to_raw_vec()));
+
+		Ok(ed25519_existing_keys
+			.chain(sr25519_existing_keys)
+			.collect::<HashSet<_>>())
```

Comment:
```suggestion
			.collect())
```

This is redundant, since compiler will figure that out from the return type.